sdk/servicebus/service-bus/review/service-bus.api.md

```diff
@@ -193,7 +193,7 @@ export interface GetMessageIteratorOptions extends OperationOptionsBase {
 
 // @public
 export interface MessageHandlers<ReceivedMessageT> {
-    processError(err: Error): Promise<void>;
+    processError(err: Error, context: ProcessErrorContext): Promise<void>;
```

Comment:
Reason to collapse this to a context vs inclining all the context info? 
If we want a context object, then why isnt the error part of the context?

Reply:
It's just mimicking event hubs:

```typescript
export type ProcessErrorHandler = (
  error: Error | MessagingError,
  context: PartitionContext
) => Promise<void>;
```

Reply:
In event hubs, the context was specific to the partition i.e [PartitionContext](https://docs.microsoft.com/en-us/javascript/api/@azure/event-hubs/partitioncontext?view=azure-node-latest) and had nothing to do with the error. The same context was passed to the callback that processes the event

Here, the context is a mix of the queue/topic/subscription/namespace details and the error source. And so my concern

Reply:
Yeah, that's an interesting point. I'm still good with this separation. The way it breaks down to me is "here's the primary thing" and "here's some optional stuff" and less which belongs to which category.

In this case the error is very much the primary piece of data.

Do you still want to change it? I'm not feeling too strongly about it, but I like the way that EventHubs breaks it down for the reasons I outlined.

Reply:
> In this case the error is very much the primary piece of data.

For me, the error source is also very much a primary piece of data. If users were to log the error message, I would advocate to log the error source as well. It helps with diagnostics. If users were to try to do different things based on the error, the error source would be important there as well.

The entity path and namespace on the other hand are secondary info. Unless one is working with multiple queues/subscriptions using the same callback, one wouldnt need to worry about this much. They can also use `reciever.fullyQualifiedNamespace` and `receiver.entityPath`

I wouldnt block the PR on this, but would open it up for a discussion with others in the JS team.

Reply:
You've all convinced me (plus @JoshLove-msft  was also making some nice use of it all). Created a ProcessErrorArgs type that contains the error and the other information that was in ProcessErrorContext.